src/serializer/ResidualHeapSerializer.js

```diff
@@ -352,75 +352,73 @@ export class ResidualHeapSerializer {
 
-      let descriptorsKey = [];
-      for (let boolKey of boolKeys) {
-        if (boolKey in desc) {
-          let b = desc[boolKey];
-          invariant(b !== undefined);
-          descProps.push(t.objectProperty(t.identifier(boolKey), t.booleanLiteral(b)));
-          descriptorsKey.push(`${boolKey}:${b.toString()}`);
-        }
-      }
+    if (!desc.get && !desc.set) {
+      boolKeys.push("writable");
+      valKeys.push("value");
+    } else {
+      valKeys.push("set", "get");
+    }
 
-      for (let descKey of valKeys) {
-        if (descKey in desc) descriptorsKey.push(descKey);
+    let descriptorsKey = [];
+    for (let boolKey of boolKeys) {
+      if (boolKey in desc) {
+        let b = desc[boolKey];
+        invariant(b !== undefined);
+        descProps.push(t.objectProperty(t.identifier(boolKey), t.booleanLiteral(b)));
+        descriptorsKey.push(`${boolKey}:${b.toString()}`);
       }
+    }
 
-      descriptorsKey = descriptorsKey.join(",");
-      let descriptorId = this.descriptors.get(descriptorsKey);
-      if (descriptorId === undefined) {
-        descriptorId = t.identifier(this.descriptorNameGenerator.generate(descriptorsKey));
-        let declar = t.variableDeclaration("var", [t.variableDeclarator(descriptorId, t.objectExpression(descProps))]);
-        // The descriptors are used across all scopes, and thus must be declared in the prelude.
-        this.prelude.push(declar);
-        this.descriptors.set(descriptorsKey, descriptorId);
-      }
-      invariant(descriptorId !== undefined);
+    descriptorsKey = descriptorsKey.join(",");
+    let descriptorId = this._descriptors.get(descriptorsKey);
+    if (descriptorId === undefined) {
+      descriptorId = t.identifier(this.descriptorNameGenerator.generate(descriptorsKey));
+      let declar = t.variableDeclaration("var", [t.variableDeclarator(descriptorId, t.objectExpression(descProps))]);
+      // The descriptors are used across all scopes, and thus must be declared in the prelude.
+      this.prelude.push(declar);
+      this._descriptors.set(descriptorsKey, descriptorId);
+    }
+    invariant(descriptorId !== undefined);
 
-      for (let descKey of valKeys) {
-        if (descKey in desc) {
-          let descValue = desc[descKey];
-          invariant(descValue instanceof Value);
-          invariant(!this.emitter.getReasonToWaitForDependencies([descValue]), "precondition of _emitProperty");
-          this.emitter.emit(
-            t.expressionStatement(
-              t.assignmentExpression(
-                "=",
-                t.memberExpression(descriptorId, t.identifier(descKey)),
-                this.serializeValue(descValue)
-              )
+    for (let descKey of valKeys) {
+      if (descKey in desc) {
+        let descValue = desc[descKey];
+        invariant(descValue instanceof Value);
+        invariant(!this.emitter.getReasonToWaitForDependencies([descValue]), "precondition of _emitProperty");
+        this.emitter.emit(
```

Comment:
Let's not do side effects here. Instead, build up a sequence expression and return that.